### PR TITLE
Dog 12 step 1

### PR DESCRIPTION
**Pull request by marqvart**

### CODE REVIEW

File: tests/test_dog_arguments.py [hunk 1]
```diff
@@ -54,63 +59,65 @@ def test_as_root(call_centos7, capfd):
...
+    # --verbose should report the actual setup
     call_centos7('--verbose', 'id')
     captured = capfd.readouterr()
     assert 'Dog Config' in captured.out
     assert "'verbose': True" in captured.out
 
 
 def test_stdin_testing_works(call_shell, capstrip):
-    '''Just verifying that my stdin testing works before testing it with dog.'''
+    # Just verifying that my stdin testing works before testing it with dog.
```
Comment (rasmus-toftdahl-olesen) [inline tests/test_dog_arguments.py:85]: Is this a new way of writing docstrings? - if that is the case then i think we should use the same syntax all over the place.

File: dog.py [hunk 2]
```diff
@@ -49,16 +51,88 @@
...
+    USER_ENV_VARS: {},
+    USER_ENV_VARS_IF_SET: {},
+    USE_PODMAN: False,
+    VERBOSE: False,
+    VERSION: DOG_VERSION,
+    VOLUMES: {},
+}
+
+
+class UsbDevices:
```
Comment (rasmus-toftdahl-olesen) [inline dog.py:95]: A shame that we have to find the USB full path - but i guess that there is no other way around.

**Review by rasmus-toftdahl-olesen — COMMENTED** [review]
Hi Lars

I like the changes - i have a few minor comments, but other than that it is good to go in my opinion.

File: dog.py [hunk 3]
```diff
@@ -437,6 +473,40 @@ def generate_env_arg_list(config: DogConfig) -> List[str]:
...
+        return await proc.wait()
+
+    loop = asyncio.get_event_loop()
+    tasks = [
+        run_volume_installer(name, image)
+        for name, image in config[VOLUMES_FROM].items()
+    ]
+    exit_codes = loop.run_until_complete(asyncio.gather(*tasks))
+    for name, ec in zip(config[VOLUMES_FROM].keys(), exit_codes):
+        if not (ec == 0 or ec == 125):
```
Comment (rasmus-toftdahl-olesen) [inline dog.py:502]: Can i get a comment about what exit code 125 means?

File: dog.py [hunk 4]
```diff
@@ -445,6 +515,8 @@ def docker_run(config: DogConfig) -> int:
     args += [
         'run',
         '--rm',
+        '--network',
```
Comment (rasmus-toftdahl-olesen) [inline dog.py:518]: I have a feeling that it is better to leave out --network unless the user specifies it explicitly in the dog.config file - what if the defaults changes in the future for some unknown reason - or is different on one platform vs. another.

**Review by rasmus-toftdahl-olesen — CHANGES_REQUESTED** [review]
Hi Lars

You are busy i can tell - i haven't read all the details - but have a few questions that i hope you can ansewr.

File: dog.py (hunk 3 above)
Comment (marqvart) [reply dog.py:502]: Yes (forgot that) - 125 is when the container (with this name) already exists. I'm actually leaning towards not checking the error code as 125 is a little too implementation specific for me - and I have not tested that it's the same using podman.

File: dog.py (hunk 4 above)
Comment (marqvart) [reply dog.py:518]: Using --network host or --network none gives a 0,5 s faster startup - it's almost half the time (!!)
I have chosen host as default (if not specified by dog.config). I'm also OK with leaving this out and only setting it if network is specified in dog.config but I felt that the speedup is something you always want.

File: dog.py (hunk 2 above)
Comment (marqvart) [reply dog.py:102]: No, I agree. This is the only fast way of translating a vendor:product to a "device" path on Linux, I could find. And that is needed as it depends on which port the device is plugged into.
An alternative would be an external script but having this as part of dog.config is really nice.

**Comment by marqvart** [comment]
All changes needed for dog 12 should now be there :-)

**Review by rasmus-toftdahl-olesen — APPROVED** [review]
I think we are there - ready for dog 12.

**Pull request merged**